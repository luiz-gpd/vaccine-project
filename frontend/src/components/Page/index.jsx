import React from 'react'
import ClayLayout from '@clayui/layout'
import VaciCuidar_logo from '../../utils/images/VaciCuidar_logo.png'

const Page = ({ title, children }) => {
    return (
        <ClayLayout.Container className="mt-4">
            <ClayLayout.Row justify="between">
                <ClayLayout.Col>
                <h1>{title}</h1>
                </ClayLayout.Col>
                <ClayLayout.Col size={3}>
                    <img src={VaciCuidar_logo} width="250px" alt="Logo do site" />
                </ClayLayout.Col>
            </ClayLayout.Row>
            { children}
        </ClayLayout.Container>
    )
}

export default Page;
