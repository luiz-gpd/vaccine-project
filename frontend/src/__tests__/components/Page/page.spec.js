import Page from "../../../components/Page"

import { render } from "@testing-library/react";

describe("Page Component", () => {
    it("renders", () => {
        const { asFragment } = render(<Page />);

        expect(asFragment()).toMatchSnapshot();
    });

    it("render with props", () => {
        const { queryByText } = render(
            <Page title="Título">
                <div className="information">
                    <span>Uno</span>
                    <span>Due</span>
                    <span>Tre</span>
                </div>
            </Page>
        );

        expect(queryByText("Título")).toBeTruthy();
        expect(queryByText("Uno")).toBeTruthy();
        expect(queryByText("Due")).toBeTruthy();
        expect(queryByText("Tre")).toBeTruthy();
    });
});
