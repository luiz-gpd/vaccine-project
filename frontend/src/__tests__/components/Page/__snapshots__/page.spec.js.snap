// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Component renders 1`] = `
<DocumentFragment>
  <div
    class="mt-4 container"
  >
    <div
      class="row justify-content-between"
    >
      <div
        class="col"
      >
        <h1 />
      </div>
      <div
        class="col-3"
      >
        <img
          alt="Logo do site"
          src="VaciCuidar_logo.png"
          width="250px"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
